<launch>

  <!-- Default argument parameters, use different values in the command line to change them -->
  <arg name="name" default="small_scout_1"/>
  <arg name="search" default="search"/>
  <arg name="mode" default="auto"/>
  <rosparam param="vol_types">["ice", "ethane", "methane", "methanol", "carbon_dioxide", "ammonia", "hydrogen_sulfite", "sulfur_dioxide", "regolith"]</rosparam>

  <group ns="$(arg name)">

    <include file="$(find obstacle)/launch/obstacle.xml">
      <arg name="name" value="$(arg name)"/>
    </include>

    <include file="$(find scoot)/launch/scoot.xml">
      <arg name="name" value="$(arg name)"/>
      <arg name="search" value="$(arg search)"/>
      <arg name="mode" value="$(arg mode)"/>
    </include>
    
    <!--
    <include file="$(find localisation)/launch/wheel_encoder.xml">
      <arg name="name" value="$(arg name)"/>
    </include>
    -->

    <!-- The stereo_image_proc node is a part of a ROS package that does stereo image processing -->
    <!-- Detailed documentation is found here: http://wiki.ros.org/stereo_image_proc -->
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen">

	<param name="prefilter_size" value="9"/>
	<param name="prefilter_cap" value="31"/>
	<param name="correlation_window_size" value="15"/>
	<param name="min_disparity" value="0"/>
	<param name="disparity_range" value="64"/>
	<param name="uniqueness_ratio" value="15.0"/>
	<param name="texture_threshold" value="10"/>
	<param name="speckle_size" value="100"/>
	<param name="speckle_range" value="4"/>
	<param name="approximate_sync" value="true"/>
	<param name="queue_size" value="5"/>

        <!-- remap default image topics to the ones used by the scout rover -->
        <remap from="left/image_raw" to="camera/left/image_raw"/>
        <remap from="right/image_raw" to="camera/right/image_raw"/>
        <remap from="left/camera_info" to="camera/left/camera_info"/>
        <remap from="right/camera_info" to="camera/right/camera_info"/>
    </node>

    <node name="drive_controller" pkg="drive_controller" type="drive_controller.py"></node>
    
  </group>
  
    <!--
  <node pkg="robot_localization" type="ekf_localization_node" name="robot_localization_ekf_node_odom" clear_params="true">
    <param name="two_d_mode" value="true"/>
    <param name="publish_tf" value="true" />
    <param name="map_frame" value="map" />
    <param name="odom_frame" value="odom" />

    <param name="base_link_frame" value="$(arg name)_tf/base_footprint"/>

    <param name="odom0" value="/small_scout_1/odom" />
    <rosparam param="odom0_config">[false, false, false,
                                    false, false, true,
                                    true, false, false,
                                    false, false, true,
                                    false, false, false]</rosparam>

    <param name="imu0" value="/small_scout_1/imu" />
    <rosparam param="imu0_config">[false, false, false,
                                   false, false, true,
                                   false, false, false,
                                   false, false, true,
                                   true, false, false]</rosparam>

     <rosparam param="initial_estimate_covariance">[1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9, 0,
                                               0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1e-9]</rosparam>

     <rosparam param="process_noise_covariance">[0.005, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0.8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0.1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0.1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0.1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0.005, 0, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0.1, 0, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0.1, 0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 1.0, 0, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1, 0, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1, 0, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1, 0, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1, 0, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1, 0,
                                                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.1]</rosparam>

    <remap from="/odometry/filtered" to="$(arg name)/odometry/filtered"/>
  </node>
   -->

</launch>
